typescript/integrations.d.ts

```diff
@@ -22,90 +22,106 @@ type IntegrationBase = {
   createdAt: Date
 }
 
-export type IntegrationTrimEnvironment = {
+export type IntegrationSecret = {
+  id: number
+  createdAt: string
+  updatedAt: string
+  cipherText: string
+  encryptionKeyId: string
+}
+
+export type SavedSecret = {
+  secretPointer: number
+}
+export type NewSecret = string
+
+export type Secret<T> = T extends SavedSecret | NewSecret ? T : SavedSecret
+
+export type IntegrationTrimEnvironment<S = SavedSecret> = {
   id: string
   label: string
   baseUrl: string
   username: string
-  password: string
+  password: Secret<S>
 }
-export type IntegrationTrim = IntegrationBase & {
+
+export type IntegrationTrim<S = SavedSecret> = IntegrationBase & {
   type: 'TRIM'
   configuration: {
-    environments: Array<IntegrationTrimEnvironment>
+    environments: Array<IntegrationTrimEnvironment<S>>
   }
 }
 
-export type IntegrationCivicaEnvironment = {
+export type IntegrationCivicaEnvironment<S = SavedSecret> = {
   id: string
   label: string
   baseUrl: string
   username: string
-  password: string
+  password: Secret<S>
 }
-export type IntegrationCivica = IntegrationBase & {
+export type IntegrationCivica<S = SavedSecret> = IntegrationBase & {
   type: 'CIVICA'
   configuration: {
-    environments: Array<IntegrationCivicaEnvironment>
+    environments: Array<IntegrationCivicaEnvironment<S>>
   }
 }
 
-export type IntegrationCPPayGateway = {
+export type IntegrationCPPayGateway<S = SavedSecret> = {
   id: string
   label: string
   clientId: string
-  clientSecret: string
+  clientSecret: Secret<S>
   paymentType?: number
 }
 
-export type IntegrationCPPay = IntegrationBase & {
+export type IntegrationCPPay<S = SavedSecret> = IntegrationBase & {
   type: 'CP_PAY'
   configuration: {
     baseUrl: string
-    gateways: IntegrationCPPayGateway[]
+    gateways: IntegrationCPPayGateway<S>[]
   }
 }
 
-export type IntegrationCPHCMS = IntegrationBase & {
+export type IntegrationCPHCMS<S = SavedSecret> = IntegrationBase & {
   type: 'CP_HCMS'
   configuration: {
     baseUrl: string
     clientId: string
-    clientSecret: string
+    clientSecret: Secret<S>
   }
 }
 
-export type IntegrationRecaptcha = IntegrationBase & {
+export type IntegrationRecaptcha<S = SavedSecret> = IntegrationBase & {
   type: 'RECAPTCHA'
   configuration: {
-    domains: IntegrationRecaptchaDomain[]
+    domains: IntegrationRecaptchaDomain<S>[]
   }
 }
 
-export type IntegrationRecaptchaDomain = {
+export type IntegrationRecaptchaDomain<S = SavedSecret> = {
   id: string
   label: string
-  privateKey: string
-  publicKey: string
+  privateKey: Secret<S>
+  publicKey: Secret<S>
```

Comment:
we dont need to encrypt this

Reply:
Is this not capable of taking actions on behalf of someone's account? ie. Pretending to be something?

Reply:
Or does it need the private key too, before it does anything?